src/fairseq2/utils/rng.py

```diff
@@ -40,3 +26,90 @@ def use_deterministic(value: bool, warn_only: bool = False) -> None:
     torch.backends.cudnn.benchmark = not value
 
     torch.use_deterministic_algorithms(value, warn_only=warn_only)
+
+
+class GeneratorBag:
+    """Holds a collection of random number generators."""
+
+    generators: List[Generator]
+
+    def __init__(self, *generators: Generator) -> None:
+        """
+        :param generators:
+            The generators to hold.
+        """
+        self.generators = list(generators)
+
+    @staticmethod
+    def from_device_defaults(*devices: Device) -> GeneratorBag:
+        """Create a :class:`GeneratorBag` instance holding the default random
+        number generators of ``devices``."""
+        generators = []
+
+        for device in set(devices):
```

Comment:
Once stored internally, the mapping is only relevant for `state_dict`/`load_state_dict`, but you are right. This might become confusing if people try to manually access `RNGBag.generators`. Let me fix this.